tests/Elasticsearch/Tests/ConnectionPool/Selectors/RoundRobinSelectorTest.php

```diff
@@ -29,19 +30,32 @@ public function testTenConnections()
     {
         $roundRobin = new Elasticsearch\ConnectionPool\Selectors\RoundRobinSelector();
 
-        $mockConnections = array();
-        foreach (range(0, 10) as $index) {
-            $mockConnections[$index] = $this->getMockBuilder('\Elasticsearch\Connections\CurlMultiConnection')
+        $mockConnections = [];
+        foreach (range(0, 9) as $index) {
+            $mockConnections[$index] = $this->getMockBuilder(ConnectionInterface::class)
                 ->disableOriginalConstructor()
                 ->getMock();
         }
 
-        foreach (range(0, 15) as $index) {
-            $retConnection = $roundRobin->select($mockConnections);
+        // select ten
+        $this->assertSame($mockConnections[0], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[1], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[2], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[3], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[4], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[5], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[6], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[7], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[8], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[9], $roundRobin->select($mockConnections));
+
+        // select five - should start from the first one (index: 0)
+        $this->assertSame($mockConnections[0], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[1], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[2], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[3], $roundRobin->select($mockConnections));
+        $this->assertSame($mockConnections[4], $roundRobin->select($mockConnections));
 
-            $nextIndex = ($index % 10) + 1;
```

Comment:
I replaced the `foreach` with explicit calls, because I believe that the original test logic was broken:
1) The first `$index` is `0`, so the mod returns `0`
2) add 1 to it
3) select `$mockConnections[1]` <- not the first connection